tt_eager/tensor/host_buffer/functions.hpp

```diff
@@ -41,6 +42,7 @@ Buffer<T> get_as(BorrowedBuffer& buffer) {
 
 template <typename T>
 Buffer<T> get_as(const BorrowedBuffer& buffer) {
+    TT_ASSERT(std::holds_alternative<Buffer<T>>(buffer), fmt::format("Unexpected type {} in {}:{} ",tt::stl::get_active_type_name_in_variant(buffer),__FILE__, __LINE__));
```

Comment:
Does it make sense to use utilities from: https://en.cppreference.com/w/cpp/utility/source_location

Reply:
Under the hood, source_location is being used by tt::stl::get_active_type_name_in_variant(buffer) when getting the type.   We could also use for file and line as well instead of the traditional `__FILE__` and `__LINE__` but if we do that we might as well swap it out with a new macro anyway so the code is more compact and less redundant.   I'm fine if you want to change this of course.